posthog/api/survey.py

```diff
@@ -973,6 +973,35 @@ def _get_survey_stats(self, date_from: str | None, date_to: str | None, survey_i
             AND event IN (%(shown)s, %(dismissed)s, %(sent)s)
             {survey_filter}
             {date_filter}
+            AND (
+                event != %(dismissed)s
+                OR
+                COALESCE(JSONExtractBool(properties, '$survey_partially_completed'), False) = False
+            )
+            AND (
+                event != %(sent)s
+                OR
+                uuid IN (
+                    (
+                        -- Select events without a submission ID (older format)
+                        SELECT uuid
+                        FROM events
+                        WHERE event = 'survey sent'
```

Comment:
i think we repeat way too many hard coded strings, how can we avoid that?
if one day we ever change an event name, a prop name, or anything, I bet we are gonna change 20 times more, its ok that we refactor file by file, PR by PR, but we are just adding more, every single PR.
https://en.wikipedia.org/wiki/Defensive_programming